src/modelmanager.cpp

```diff
@@ -209,7 +210,26 @@ void processPipelineConfig(rapidjson::Document& configJson, const rapidjson::Val
         nodeName = nodeConfig["name"].GetString();
 
         std::string modelName;
-        modelName = nodeConfig["model_name"].GetString();
+        if (nodeConfig.HasMember("model_name")) {
+            modelName = nodeConfig["model_name"].GetString();
+        }
+
+        NodeLibrary nodeLibrary;
+        if (nodeConfig.HasMember("library_name")) {
+            std::string libraryName = nodeConfig["library_name"].GetString();
+            auto status = manager.getCustomNodeLibraryManager().getLibrary(libraryName, nodeLibrary);
+            if (!status.ok()) {
+                SPDLOG_LOGGER_WARN(modelmanager_logger, "Pipeline: {} refers too non existing custom node library: {}", pipelineName, libraryName);
+                return;
+            }
+        }
```

Comment:
Shouldn't it be:
if (model_name)
else if (library_name)
else (SomeRetCode)?

Even better it could decide on the type of node what to look for, and you could have separate function for parsing model type node & custom node type to shorten up this already long function.

Reply:
Done